src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -31,14 +31,12 @@ internal class SslStreamInternal
         private int _nestedRead;
         private AsyncProtocolRequest _readProtocolRequest; // cached, reusable AsyncProtocolRequest used for read operations
         private AsyncProtocolRequest _writeProtocolRequest; // cached, reusable AsyncProtocolRequest used for write operations
+        private Action<Task, object> _freeBufferAction;
```

Comment:
Can you explain how the buffer management has changed here?  It seems like we used to track the buffer itself (_pinnableOutputBuffer), and now we have this delegate to manage the buffer free instead.  What's the reason for this change?